### PR TITLE
Add option to commit changes to the database automatically

### PR DESCRIPTION
**Pull request by Azrenbeth**
BUILDS ON azren/early_abort_negative_compression

This adds the -c option which commits the changes to the database automatically
(this saves you from having to run it as a seperate command and allows easier testing)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -358,11 +371,94 @@ pub fn run(mut config: Config) {
...
+                        PGEscape(e)
+                    ));
+                }
+                sql.push_str(";\n");
+            }
+
+            return sql;
+        }
+        "".to_string()
+    })
```
Comment (erikjohnston) [inline src/lib.rs:459]:
```suggestion
            sql
        } else {
            String::new()
        }
    })
```

**Comment by erikjohnston** [comment]
I need to have a closer look at stuff tomorrow, but at a high level looks good

File: src/database.rs [hunk 2]
```diff
@@ -339,3 +341,104 @@ fn test_pg_escape() {
...
+    config: &Config,
+    old_map: &BTreeMap<i64, StateGroupEntry>,
+    new_map: &BTreeMap<i64, StateGroupEntry>,
+) {
+    // connect to the database
+    let mut builder = SslConnector::builder(SslMethod::tls()).unwrap();
+    builder.set_verify(SslVerifyMode::NONE);
+    let connector = MakeTlsConnector::new(builder.build());
+
+    let mut client = Client::connect(&config.db_url, connector).unwrap();
```
[the remark this answers is not in the document]
Comment (Azrenbeth) [reply src/database.rs:365]: I guess we could, however if compression takes a long time, then that connection isn't getting used for a long time. Is that some thing that postgres would be ok with?

**Review by erikjohnston — APPROVED** [review]
LGTM!

File: src/database.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Azrenbeth) [reply src/database.rs:365]: Leaving as future possible TODO

**Pull request merged**